Plugins/Public/dronebays/UserCommands.cpp

```diff
@@ -0,0 +1,377 @@
+
+	// Verify that the client isn't cruising or in a tradelane
+	const ENGINE_STATE engineState = HkGetEngineState(iClientID);
+	if(engineState == ES_TRADELANE || engineState == ES_CRUISE)
+	{
+		PrintUserCmdText(iClientID, L"Engine state inoppertune for drone deployment, aborting launch");
+		return true;
+	}
+
+	//Get the drone type argument - We don't care about any garbage after the first space 
+	const string reqDroneType = wstos(GetParam(wscParam, L' ', 0));
+
+	// Verify that the requested drone type is a member of the bay's available drones
+	if(availableDroneArch.find(reqDroneType) == availableDroneArch.end())
+	{
+		PrintUserCmdText(iClientID, L"Your drone bay does not support this type of deployment");
+		return true;
+	}
+
+	const DroneArch requestedDrone = availableDroneArch[reqDroneType];
+
+	// All of the required information is present! Build the timer struct and add it to the list
+	DroneBuildTimerWrapper wrapper;
+	wrapper.buildTimeRequired = clientDroneInfo[iClientID].droneBay.iDroneBuildTime;
+	wrapper.reqDrone = requestedDrone;
+	wrapper.startBuildTime = timeInMS();
+
+	buildTimerMap[iClientID] = wrapper;
+
+	// Set the buildstate, and alert the user
+	clientDroneInfo[iClientID].buildState = STATE_DRONE_BUILDING;
+	PrintUserCmdText(iClientID, L"Drone being prepared for deployment :: Launch in ETA %i seconds", clientDroneInfo[iClientID].droneBay.iDroneBuildTime);
+
+	// Save the carrier shipObj to the client struct
+	clientDroneInfo[iClientID].carrierShipobj = playerShip;
+	return true;
+}
+
+bool UserCommands::UserCmd_AttackTarget(uint iClientID, const wstring& wscCmd, const wstring& wscParam,
+	const wchar_t* usage)
+{
+	// Verify that the user is in space
+	uint iShipObj;
+	pub::Player::GetShip(iClientID, iShipObj);
+	if(!iShipObj)
+	{
+		PrintUserCmdText(iClientID, L"You must be in space to use this command");
+		return true;
+	}
+
+	// Verify that the user has a drone currently deployed
+	if(clientDroneInfo[iClientID].deployedInfo.deployedDroneObj == 0)
+	{
+		PrintUserCmdText(iClientID, L"You must have a drone deployed for this to work");
+		return true;
+	}
+
+	// Get the players current target
+	uint iTargetObj;
+	pub::SpaceObj::GetTarget(iShipObj, iTargetObj);
+
+	if(!iTargetObj)
+	{
+		PrintUserCmdText(iClientID, L"Please target the vessel which the drones should be directed to");
+		return true;
+	}
+
+	const uint droneObj = clientDroneInfo[iClientID].deployedInfo.deployedDroneObj;
+
+	// Set the old target to neutral reputation, and the hostile one hostile.
+	Utility::SetRepNeutral(droneObj, clientDroneInfo[iClientID].deployedInfo.lastShipObjTarget);
+	Utility::SetRepHostile(droneObj, iTargetObj);
+
+	// Set the target hostile to the drone as well only if it isn't another existing drone 
+	bool isTargetDrone = false;
+	for (const auto& currClient : clientDroneInfo)
+	{
+		if (iTargetObj == currClient.second.deployedInfo.deployedDroneObj)
+		{
+			isTargetDrone = true;
+			break;
+		}
+	}
+
+	if (!isTargetDrone)

```

Comment:
what should we do in the opposite case?

Reply:
The code below the if statement seems valid to me.